### DISCUSSION TITLE
Hot to access to x-data of an element from external js

### DISCUSSION THREAD

**Discussion by neapolis79** [alpinejs/alpine#1763] [1. Help] [answered]
Hi 
recently i'm verifying if my application can be compatible also with Aplinejs v3. In my project with Alpine v2 i can read and write x-data parameter using 
`document.getElementById('<element>).__x.$data`
but in v3 this doesn't works anymore....how can i r/w object x-data from a a simple script javascript?

Thanks in advance

**Comment by Pronian** [comment, marked as answer]
The v3 replacement for `document.getElementById('<element>).__x.$data` is `document.getElementById('<element>)._x_dataStack[0]` but keep in mind that this is not the recommended way of getting data from a component. The recommended way is either using $store or via custom events with $dispatch
